pkg/gui/controllers/list_controller.go

```diff
@@ -53,13 +53,19 @@ func (self *ListController) HandleScrollRight() error {
 func (self *ListController) HandleScrollUp() error {
 	scrollHeight := self.c.UserConfig.Gui.ScrollHeight
 	self.context.GetViewTrait().ScrollUp(scrollHeight)
+	if self.context.RenderOnlyVisibleLines() {
+		return self.context.HandleRender()
+	}
```

Comment:
It's unfortunate that we need to do this manually here (and in HandleFocus, see previous commit). It would be nice if we had a way to automatically call it whenever the origin changed (similar to what we do for the height in layout), but I haven't found a good way to do that.